Lib/fontTools/merge.py

```diff
@@ -941,6 +941,38 @@ def __delitem__(self, k):
 		del self.d[id(k)]
 
 class Merger(object):
+	"""Font merger.
+
+	This class merges multiple files into a single OpenType font, taking into
+	account complexities such as OpenType layout (``GSUB``/``GPOS``) tables and
+	cross-font metrics (e.g. ``hhea.ascent`` is set to the maximum value across
+	all the fonts).
+
+	If multiple glyphs map to the same Unicode value, subsequent glyphs are
+	renamed and a lookup in the ``locl`` feature will be created to disambiguate
+	them. For example, if the arguments are an Arabic font and a Latin font and
+	both contain a set of parentheses, the Latin glyphs will be renamed to
+	``parenleft#1`` and ``parenright#1``, and the equivalent of the following
```

Comment:
The addition of hash characters to disambiguate glyph names in the merger reminds me of https://github.com/fonttools/fonttools/issues/1942

Unlike with that other issue (where a font contains duplicates in post table that fonttools needs to deduplicate internally and temporarily), in the case of the merger, we _are_ writing invalid glyph names (containing an invalid character in postscript `#`) in the merged font's `post` table.

Let's open a separate issue for this; maybe we can us a dot `.` here.

/cc @justvanrossum @jenskutilek

Reply:
Note that there is also code that checks if the duplicate glyphs are the same, so there are more criteria considered than code point: https://github.com/fonttools/fonttools/blob/cfb261cf0279c352538f06f191c2df751477a177/Lib/fontTools/merge.py#L368-L377